packages/cli/src/api/formats/fixtures/messages_plural.po

```diff
@@ -0,0 +1,32 @@
+msgid ""
+msgstr ""
+"POT-Creation-Date: 2018-08-27 10:00+0000\n"
+"Mime-Version: 1.0\n"
+"Content-Type: text/plain; charset=utf-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"X-Generator: @lingui/cli\n"
+"Language: en\n"
+
+msgctxt "pluralize_on=someCount"
+msgid "message_with_id"
+msgid_plural "message_with_id_plural"
+msgstr[0] "Singular case"
+msgstr[1] "Case number {someCount}"
+
+msgctxt "icu=%7BanotherCount%2C+plural%2C+one+%7BSingular+case%7D+other+%7BCase+number+%7BanotherCount%7D%7D%7D&pluralize_on=anotherCount"
+msgid "Singular case"
+msgid_plural "Case number {anotherCount}"
+msgstr[0] "Singular case"
+msgstr[1] "Case number {anotherCount}"
+
+msgctxt "pluralize_on=count"
+msgid "message_with_id_but_without_translation"
+msgid_plural "message_with_id_but_without_translation_plural"
+msgstr[0] ""
+msgstr[1] ""
+
+msgctxt "icu=%7Bcount%2C+plural%2C+one+%7BSingular%7D+other+%7BPlural%7D%7D&pluralize_on=count"
```

Comment:
Is it necessary for this feature to use `msgctx`? There's actually another PR #856 implementing the original `msgctx` behavior from Gettext.

Reply:
Storing the original ICU message in `msgctx` is clearly a workaround and not using the field as intended. In earlier versions, only the `pluralize_on` value was stored there, which I need to reconstruct the original ICU message.
The full ICU message is stored s.t. `msgid` can be restored for messages where the developer does not use custom IDs, as the ICU cases in development language are used for `msgid` and `msgid_plural`, so items look like this:

```po
msgctxt "icu=%7Bcount%2C+plural%2C+one+%7BSingular%7D+other+%7BPlural%7D%7D&pluralize_on=count"
msgid "Singular"
msgid_plural "Plural"
msgstr[0] ""
msgstr[1] ""
```

I could also store the querystring encoded data in a new type of comment, say ' #?foo=bar' and, when converting from po to ICU, iterate comments until I find one that matches the format. What do you think about that?

Reply:
Yeah, I think we need to use comments to store any required metadata. You could even have one comment per line:

```po
#. icu: { count, plural, one {Singular} other {Plural}}
#. pluralize_on: count
msgid "Singular"
msgid_plural "Plural"
msgstr[0] ""
msgstr[1] ""
```

but whatever works for you the best 👍 

There're several types of comments available in [gettext](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html):

```
white-space
#  translator-comments
#. extracted-comments
#: reference…
#, flag…
#| msgid previous-untranslated-string
msgid untranslated-string
msgstr translated-string
```

Not sure which are supported by the PO library we use, but I guess this would be a good fit for `extracted-comments`.

I'm open to any suggestions :)